templates/skeleton/daml.yaml.template

```diff
@@ -12,3 +12,18 @@ dependencies:
   - daml-script
 sandbox-options:
   - --wall-clock-time
+
+# https://discuss.daml.com/t/making-the-most-out-of-daml-compiler-warnings/739
+build-options: [
+  # make compilation fail if there is a warning
+  #"--ghc-option", "-Werror",
+
+  # turn on all warnings; this may be too verbose
+  #"--ghc-option", "-Wall",
+
+  "--ghc-option", "-Wunused-matches",
+  "--ghc-option", "-Wunused-do-bind",
```

Comment:
Maybe only include one of the 5 to make it a bit less noisy? `-Wunused-matches` seems like a good choice.

Reply:
I'm not sure I agree with that as none of them are noisy when starting, and it's a lot easier to have them from the start. I've commented the other 4.

Reply:
Sorry I didn’t mean that they are noisy in the sense that they produce a lot of warnings but in the sense that it adds noise to the `daml.yaml` file. Think of someone relatively new to DAML that now looks at this file.

With this change  half of this file is devoted to the warning section. If you ask experienced DAML developers at DA, I expect most of them are not going to be able to tell you what any of these warnings do and even in our team, I wouldn’t be surprised if we end up with a 50% split of people that understand all these warnings. `-Wmissing-signatures` further has issues in that it interacts badly with the return-type feature we have in DAML and `-Wredundant-constraints` used to be somewhat broken but that might have been fixed in GHC.

I think it makes sense to show people that there are warnings they can turn on and maybe enable one by default to show the effect and the syntax but I don’t think it makes sense to devote half of the default `daml.yaml` to warnings they don’t understand.

Reply:
I see. I guess that's a matter of opinion; I am quite fond of programs that come with a default config file that includes every single possible option, with its value set to its default, and a comment explaining what the option does and what other values it can take.

I'll admit it can feel a bit overwhelming at first but it also makes it very easy to find out what can be configured and what the syntax is. I would want that in the skeleton (though not in, say, create-daml-app).

Perhaps this should be a separate conversation though.

Reply:
I suppose it boils down to defining the use-case for the skeleton template: is this meant for the first project of a new user, or is this meant as a template for new projects all throughout a DAMLer's life? If the former I agree we should limit noise; if the latter I think we should have all the warnings we recommend enabled, and seriously consider adding all the possible options so people can quickly scan through and set the ones they want when starting a new project.